kotlin/injivcrenderer/src/main/java/io/mosip/injivcrenderer/InjiVcRenderer.kt

```diff
@@ -0,0 +1,51 @@
+package io.mosip.injivcrenderer
+import io.mosip.injivcrenderer.DateUtils.formatDate
+import io.mosip.injivcrenderer.DateUtils.isValidDateTime
+import io.mosip.injivcrenderer.NetworkHelper.fetchSvgAsText
+import okio.IOException
+import org.json.JSONObject
+
+class InjiVcRenderer {
+
+    // Method to get value from nested JSON data
+    fun getValueFromData(key: String, data: JSONObject): Any? {
+        val keys = key.split("/")
+        var value: Any? = data
+        for (k in keys) {
+            if (value is JSONObject) {
+                value = value.opt(k)
+            } else {
+                return null
+            }
+        }
+        return value
+    }
+
+    // Method to replace placeholders in the SVG template
+    fun renderSvg(vcJsonData: String): String {
+        try {
+            val jsonObject = JSONObject(vcJsonData)
+            val renderMethodArray = jsonObject.getJSONArray("renderMethod")
```

Comment:
renderMethodArray can be null

Reply:
Only if the VC has renderMethod , we are calling the library from client.